server/src/main/java/org/opensearch/action/admin/cluster/stats/ClusterStatsResponse.java

```diff
@@ -131,8 +163,13 @@ public void writeTo(StreamOutput out) throws IOException {
         out.writeVLong(timestamp);
         out.writeOptionalWriteable(status);
         out.writeOptionalString(clusterUUID);
-        out.writeOptionalWriteable(indicesStats.getMappings());
-        out.writeOptionalWriteable(indicesStats.getAnalysis());
+        if (indicesStats != null) {
+            out.writeOptionalWriteable(indicesStats.getMappings());
+            out.writeOptionalWriteable(indicesStats.getAnalysis());
+        } else {
+            out.writeOptionalWriteable(null);
+            out.writeOptionalWriteable(null);
```

Comment:
did you test, this is working as expected and not causing serialization issue?

Reply:
yes, verified this.